src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -1,27 +1,46 @@
 package mate.academy.spring.config;
 
+import mate.academy.spring.model.Role;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private static final String ADMIN = Role.Type.ADMIN.name();
+    private static final String USER = Role.Type.USER.name();
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+
+    public SecurityConfig(UserDetailsService userDetailsService, PasswordEncoder passwordEncoder) {
+        this.userDetailsService = userDetailsService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
     @Autowired
     public void configure(AuthenticationManagerBuilder auth) throws Exception {
-        auth
-                .inMemoryAuthentication()
-                .passwordEncoder(getEncoder())
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .antMatchers(HttpMethod.POST,"/register").permitAll()
+                .antMatchers(HttpMethod.GET,"/cinema-halls",
+                        "/movies","/movie-sessions/available").hasAnyRole(ADMIN,USER)
+                .antMatchers(HttpMethod.POST,"/cinema-halls",
+                        "/movies","/movie-sessions").hasRole(ADMIN)
```

Comment:
I think yiu should add / to urls, because without it you will have error 403 Forbidden when you will try get or post some data
```suggestion
                .antMatchers(HttpMethod.GET,"/cinema-halls/",
                        "/movies/","/movie-sessions/available").hasAnyRole(ADMIN,USER)
                .antMatchers(HttpMethod.POST,"/cinema-halls/",
                        "/movies/","/movie-sessions/").hasRole(ADMIN)
```